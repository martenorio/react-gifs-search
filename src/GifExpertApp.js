import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
export const GifExpertApp = ({ defaultCategories = [] }) => {
    // const [categories, setCategories] = useState(['trending']);
    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {
    //     //setCategories([ ...categories, 'categories ' + (categories.length + 1)])
    //     setCategories( cate => [ ...cate, 'categories ' + (cate.length + 1)])
    // }

    return (
        <>
            <h2>Gif expert app</h2>
            <AddCategory setCategories={setCategories}></AddCategory>
            <hr></hr>
            {
                categories.map(category => {
                    return <GifGrid key={category} category={category}></GifGrid>
                })
            }
        </>
    )
}