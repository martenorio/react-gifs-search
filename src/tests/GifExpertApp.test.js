import React from 'react'
import { GifExpertApp } from "../GifExpertApp";
import { shallow } from 'enzyme';

describe('Test GifExperApp', () => {
    
    test('should show <GifExperApp />', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    })
    test('should show categories list', () => {
        const categories = ['day','night'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);
        // console.log(wrapper.find('GifGrid').length);
        // console.log(categories.length);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
})
