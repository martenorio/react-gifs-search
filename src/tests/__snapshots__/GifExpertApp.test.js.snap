// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifExperApp should show <GifExperApp /> 1`] = `
<Fragment>
  <h2>
    Gif expert app
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Test GifExperApp should show categories list 1`] = `
<Fragment>
  <h2>
    Gif expert app
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="day"
    key="day"
  />
  <GifGrid
    category="night"
    key="night"
  />
</Fragment>
`;
